gulpfile.js

```diff
@@ -21,12 +24,35 @@ var sources = [
 ].map(function (tsFolder) { return tsFolder + '/**/*.ts'; })
     .concat(['test/*.ts']);
 
+var knownOptions = {
+  string: 'env',
+  default: { env: 'production' }
+};
+
+var options = minimist(process.argv.slice(2), knownOptions);
+
+function getArgumentIndex(argumentName) {
+    return process.argv.indexOf("--" + argumentName);
+}
+
+function readArgument(argumentName) {
```

Comment:
How about using this instead of parsing manually? http://stackoverflow.com/questions/23023650/is-it-possible-to-pass-a-flag-to-gulp-to-have-it-run-tasks-in-different-ways